app/src/main/java/br/com/catbag/gifreduxsample/asyncs/data/storage/Database.java

```diff
@@ -1,19 +1,53 @@
 package br.com.catbag.gifreduxsample.asyncs.data.storage;
 
+import android.content.Context;
+import android.util.Log;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.snappydb.DB;
+import com.snappydb.DBFactory;
+import com.snappydb.SnappydbException;
+
+import java.io.IOException;
 import java.util.ArrayList;
 import java.util.List;
 import java.util.UUID;
 
+import br.com.catbag.gifreduxsample.models.AppState;
 import br.com.catbag.gifreduxsample.models.Gif;
+import br.com.catbag.gifreduxsample.models.ImmutableAppState;
 import br.com.catbag.gifreduxsample.models.ImmutableGif;
 
+import static br.com.catbag.gifreduxsample.helpers.AppStateHelper.gifListToMap;
+
 /**
  * Created by felipe on 26/10/16.
  */
 
 public class Database {
```

Comment:
I would recommends to write simple unit tests for those methods, only to ensures the right behavior.